File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/ImageInfoImpl.kt
package dev.hebei.camerax_android.core

import androidx.camera.core.ImageInfo
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiImageInfoApi

class ImageInfoImpl(impl: CameraXImpl) : PigeonApiImageInfoApi(impl) {
    override fun timestamp(pigeon_instance: ImageInfo): Long {
        return pigeon_instance.timestamp
    }

    override fun rotationDegrees(pigeon_instance: ImageInfo): Long {
        return pigeon_instance.rotationDegrees.toLong()
    }
}

File: camerax_ios/ios/Classes/core/ImageInfoImpl.swift
//
//  ImageInfoImpl.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/16.
//

import Foundation

class ImageInfoImpl: PigeonApiDelegateImageInfoApi {
    func timestamp(pigeonApi: PigeonApiImageInfoApi, pigeonInstance: ImageInfo) throws -> Int64 {
        return Int64(pigeonInstance.timestamp)
    }
    
    func rotationDegrees(pigeonApi: PigeonApiImageInfoApi, pigeonInstance: ImageInfo) throws -> Int64 {
        return Int64(pigeonInstance.rotationDegrees)
    }
}
